### PR TITLE
User Agent implementation

### PR DESCRIPTION
**Pull request by ASypula**

### CODE REVIEW

File: src/agents/user_agent.py [hunk 1]
```diff
@@ -1,12 +1,126 @@
...
+                body=json.dumps(check_offers_msg.dict()),
+                metadata={"performative": "query-ref", "action": "check-offers"}
+            )
+            await self.send(to_send)
+
+    def choose_parking_offer(self, offers: List[dict]):
+        #TODO implement logic of choosing an offer
+        return offers[0]["parking_id"] if offers else None
+
+    class MakeReservation(CyclicBehaviour):
```
Comment (WasilewM) [inline src/agents/user_agent.py:78]: Is this a `CyclicBehaviour` or an `OneShotBehaviour`?

Comment (WasilewM) [inline src/agents/user_agent.py:78]: Is this a `CyclicBehaviour` or an `OneShotBehaviour`?

File: src/agents/user_agent.py [hunk 2]
```diff
@@ -1,12 +1,126 @@
...
+    class AwaitReservationConfirmation(CyclicBehaviour):
+        """Behaviour for waiting for reservation confirmation"""
         async def run(self):
-            # @TODO to be implemented
-            pass
+            msg = await self.receive(timeout=RECEIVE_TIMEOUT)
+            if msg:
+                reservation_response = ReservationResponse(**json.loads(msg.body))
+                if reservation_response.success:
+                    #TODO What about reservation id?
```
Comment (WasilewM) [inline src/agents/user_agent.py:101]: How about a timestamp + random number for reservation id? Or simply a random number.

Comment (ASypula) [reply src/agents/user_agent.py:101]: I guess the reservation ids should be unique across all users. I was thinking about making a hash using the user's id and maybe timestamp?

File: src/agents/user_agent.py (hunk 1 above)
Comment (szaryvip) [reply src/agents/user_agent.py:78]: As I understand from documentation, OneShotBehaviour could be realized only one time in agent life, so if we would like to make multiple reservations, we should have this behaviour as CyclicBehaviour

**Comment by szaryvip** [comment]
I think, we are missing Modifying Reservation

**Comment by ASypula** [comment]
> I think, we are missing Modifying Reservation

Yep, will be adding that, for now is WIP but I wanted to check with you if the baseline for UserAgent is ok. Thanks!

**Review by WasilewM — COMMENTED** [review]
Please update changelog

**Comment by ASypula** [comment]
> Please update changelog

Is it ok now? Or would you prefer more details?

**Review by WasilewM — APPROVED** [review]

**Pull request merged**